src/node_report.cc

```diff
@@ -635,35 +716,47 @@ static void PrintVersionInformation(FILE* fp, Isolate* isolate) {
  * Function to print the JavaScript stack, if available
  *
  ******************************************************************************/
-static void PrintJavaScriptStack(FILE* fp, Isolate* isolate, DumpEvent event, const char* location) {
-  fprintf(fp, "\n================================================================================");
-  fprintf(fp, "\n==== JavaScript Stack Trace ====================================================\n\n");
+static void PrintJavaScriptStack(std::ostream& out, Isolate* isolate, DumpEvent event, const char* location) {
+  out << "\n================================================================================";
+  out << "\n==== JavaScript Stack Trace ====================================================\n\n";
 
 #ifdef _WIN32
   switch (event) {
   case kFatalError:
     // Stack trace on fatal error not supported on Windows
-    fprintf(fp, "No stack trace available\n");
+    out << "No stack trace available\n";
     break;
   default:
     // All other events, print the stack using StackTrace::StackTrace() and GetStackSample() APIs
-    PrintStackFromStackTrace(fp, isolate, event);
+    PrintStackFromStackTrace(out, isolate, event);
     break;
   }  // end switch(event)
 #else  // Unix, OSX
   switch (event) {
   case kException:
-  case kJavaScript:
+  case kJavaScript: {
     // Print the stack using Message::PrintCurrentStackTrace() API
-    Message::PrintCurrentStackTrace(isolate, fp);
+    std::FILE *stack_fp = std::tmpfile();
+    if (stack_fp != nullptr) {
+      char stack_buf[64];
+      Message::PrintCurrentStackTrace(isolate, stack_fp);
+      std::fflush(stack_fp);
+      std::rewind(stack_fp);
+      while (std::fgets(stack_buf, sizeof(stack_buf), stack_fp) != nullptr) {
+        out << stack_buf;
+      }
+      // Calling close on a file from tmpfile *should* delete it.
+      std::fclose(stack_fp);
+    }
```

Comment:
I suppose there's no fallback here if `std::tmpfile()` fails? Should we emit some sort of message, e.g. "Unable to obtain stack trace"?